plugins/action/common/prepare_plugins/prep_107_policy.py

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) 2024 Cisco Systems, Inc. and its affiliates
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of
+# this software and associated documentation files (the "Software"), to deal in
+# the Software without restriction, including without limitation the rights to
+# use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+# the Software, and to permit persons to whom the Software is furnished to do so,
+# subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+# FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# SPDX-License-Identifier: MIT
+
+
+class PreparePlugin:
+    def __init__(self, **kwargs):
+        self.kwargs = kwargs
+        self.keys = []
+
+    def prepare(self):
+        model_data = self.kwargs['results']['model_extended']
+
+        # import epdb; epdb.st()
+
+        # Ensure that vrf_lite's switches are mappinged to their respective
+        # management IP address from topology switches
+        topology_switches = model_data['vxlan']['topology']['switches']
+        policy_switches = model_data['vxlan']['policy']['switches']
+        for switch in policy_switches:
+            if any(sw['name'] == switch['name'] for sw in topology_switches):
+                found_switch = next((item for item in topology_switches if item["name"] == switch['name']))
+                if found_switch.get('management').get('management_ipv4_address'):
+                    switch['name'] = found_switch['management']['management_ipv4_address']
+                elif found_switch.get('management').get('management_ipv6_address'):
+                    switch['name'] = found_switch['management']['management_ipv6_address']
+
+        self.kwargs['results']['model_extended'] = model_data
```

Comment:
I don't see where you are making any modifications to the extended model_data before you pass it back to the ansible playbook.  Not sure I follow how this is being used.

Reply:
It should be just a mutable object reference getting updated if not mistaken.

The data model looks like this:

```
---
vxlan:
  policy:
    <snip>
    switches:
      - name: 10.15.33.13
        groups:
          - server leafs
      - name: netascode-leaf4
        groups:
          - server leafs
```

Post the prepare plugin, the data looks like this:

```
{
    "vxlan": {
        "policy": {
            <snip>
            "switches": [
                {
                    "groups": [
                        "server leafs"
                    ],
                    "name": "10.15.33.13"
                },
                {
                    "groups": [
                        "server leafs"
                    ],
                    "name": "10.15.33.16"
                }
            ]
        }
    }
}
```

I can try to do this differently if you think it's needed.

Reply:
Chatted about this and I am fine with the changes